src/Microsoft.Sbom.Api/Utils/Comparer/SbomRelationshipComparer.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Sbom.Contracts;
+
+namespace Microsoft.Sbom.Api.Utils.Comparer;
+
+/// <summary>
+/// Compares SBOM relationships based on their properties.
+/// </summary>
+public class SbomRelationshipComparer : IEqualityComparer<SbomRelationship>
+{
+    public bool Equals(SbomRelationship relationship1, SbomRelationship relationship2)
+    {
+        if (relationship1 == null || relationship2 == null)
```

Comment:
I think we're trying to use `is null` and `is not null` instead of `== null` and `!= null` in new code